x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/nulls/CoalesceTests.java

```diff
@@ -45,12 +51,67 @@ public CoalesceTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testC
      */
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
+        List<TestCaseSupplier> suppliers = new ArrayList<>();
         VaragsTestCaseBuilder builder = new VaragsTestCaseBuilder(type -> "Coalesce");
         builder.expectString(strings -> strings.filter(v -> v != null).findFirst());
         builder.expectLong(longs -> longs.filter(v -> v != null).findFirst());
         builder.expectInt(ints -> ints.filter(v -> v != null).findFirst());
         builder.expectBoolean(booleans -> booleans.filter(v -> v != null).findFirst());
-        return parameterSuppliersFromTypedData(builder.suppliers());
+        suppliers.addAll(builder.suppliers());
+        addSpatialCombinations(suppliers);
+        suppliers.add(
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.IP),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("192.168.0.10"), DataTypes.IP, "first"),
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("192.168.0.20"), DataTypes.IP, "second")
+                    ),
+                    "CoalesceEvaluator[values=[Attribute[channel=0], Attribute[channel=1]]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("192.168.0.10"))
```

Comment:
It'd be nice if these were random, but I can live with it either way. It'd also be nice to have a case where the first is null too, if we're here.